doc/developer/project-management.md

```diff
@@ -0,0 +1,122 @@
+unbounded as making sure that users can sufficiently observe Materialize’s
+performance. Both types of problems are important to solve in order to
+ensure the overall success of Materialize as a product.
+
+By default, all created issues will end up in a Materialize-wide
+backlog. Issues created in our public Github repository will be
+visible outside of our team.
+
+## Triage
+
+The purpose of the Triage phase is to ensure we’re working on the
+most urgent and important problems at all times.
+
+Newly reported bugs will be triaged most frequently. `P0` or `release-blocker`
+bugs will be triaged within 24 hours, and all other bugs will be
+triaged within 3 business days. The remaining backlog of problem
+Issues will be triaged by EPD management at least once a month.
+
+Some problems will be sufficiently straightforward with a clearly
+defined scope, a limited solution space, and limited effects on other
+teams and projects. In these cases, these Github Issues can be directly
+assigned to an owner for implementation.
+
+Other problems will be less straightforward. Some may require more
+digging or debugging to understand what’s going wrong. Some may only
+highlight a symptom of a larger issue with the product, and require
+the Product team to discover the true underlying problem. Some may
+impact the work of other teams or require ongoing stakeholder management.
+In any of these cases, or in any case where a solution does not feel
+simple, problems should be turned into a Github Epic.
+
+Each standalone Issue and Epic will be prioritized and tagged with
+a Milestone: [“Now,” “Next,” or “Later,”](https://www.notion.so/Product-Planning-and-Prioritization-Guidance-ce7f91bd5e224a71841630c306414700#b745adb7b4604268a93033b38a504aae)
+indicating when work for the project will begin. A DRI must be
+assigned to an Issue or an Epic as it is marked to be worked on
+“Now,” moving the project forward to the Discovery phase.
+
+## Discovery
+
+The purpose of the Discovery phase is to sufficiently explore the
+solution space of a problem before committing to a particular solution.
+Design documents are the tool that we use to do this. The DRI is
+responsible for driving the Design Document Process for their project.
+
+If a project is sufficiently straightforward, the DRI can opt to skip
+the Design Document Process. The DRI will need to write that they’re
+skipping this step in the Github Epic.
+
+Otherwise, the DRI is responsible for researching the solution space,
+proposing potential solutions, and getting signoff from relevant stakeholders.
+This does not mean that DRI is required to convince everyone of their chosen
+solution, we do not intend to design by committee. If you are the DRI for a
+project and need help identifying your relevant stakeholders, please reach
+out to your manager.
+
+## Delivery
+
+Once a design has been approved, the DRI can move a project from Discovery
+to Delivery. During Delivery, the DRI is responsible for ushering
```

Comment:
I'm not sure if this will be successful. For many changes, we need to work on the design in parallel to the implementation because there is no other way to figure out what question the design doc needs to answer. How does this fit within this proposal?

Reply:
The discovery phase explicitly asks for prototypes. It's totally fine if they evolve together with the design. I think your work on unbilled replicas is a good example. The design and prototype evolved hand in hand and once the design is approved you will deliver based on the prototype – it's totally fine if the diff between the prototype and the final delivery is small.